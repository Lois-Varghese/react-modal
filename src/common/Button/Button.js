import React from 'react';
import './Button.scss';

const Button = ({children, onClick, ...props}) => {
  return (
    <button type="button" onClick={onClick} className="btn" {...props}>
      {children}
    </button>
  );
};

export default Button;
