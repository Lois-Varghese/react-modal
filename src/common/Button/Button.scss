@import '../../styles/abstracts/variables';

.btn {
  font-family: $font_family_inter;
  position: relative;
  font-weight: $font_weight_400;
  font-size: $medium_font;
  line-height: 2;
  height: 40%;
  transition: all 200ms linear;
  border-radius: 4px;
  width: 70%;
  letter-spacing: 1px;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  align-self: center;
  border: none;
  cursor: pointer;
  background-color: $color_light_white;
  color: $color_red;
  box-shadow: 0 12px 35px 0 $color_black;
  outline: 0;
  text-transform: capitalize;
}

.btn:hover {
  background-color: $color_red;
  color: $color_light_white;
}

@media screen and (max-width: 768px) {
  .btn {
    font-family: $font_family_inter;
    position: relative;
    font-weight: $font_weight_400;
    font-size: $small_font;
    line-height: 1;
    height: 30%;
    transition: all 200ms linear;
    border-radius: 4px;
    width: 60%;
    margin-bottom: 20px;
    letter-spacing: 1px;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    align-self: center;
    border: none;
    cursor: pointer;
    background-color: $color_light_white;
    color: $color_red;
    box-shadow: 0 12px 35px 0 $color_black;
    outline: 0;
    text-transform: capitalize;
  }

  .btn:hover {
    background-color: $color_red;
    color: $color_light_white;
  }
}
