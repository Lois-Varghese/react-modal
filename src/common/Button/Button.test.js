import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Button from "./Button";

configure({ adapter: new Adapter() });
const wrapper = shallow(<Button />);
let container, containerProp;

describe("Render component", () => {
  it("should render Button Component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("render <button>", () => {
  beforeEach(() => {
    container = wrapper.find("button");
    containerProp = container.props();
  });

  it("should have a <button> with className", () => {
    expect(containerProp.className).toEqual("btn");
  });
});
