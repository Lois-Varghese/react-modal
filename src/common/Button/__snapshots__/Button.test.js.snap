// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component should render Button Component 1`] = `
<button
  className="btn"
  type="button"
/>
`;
